cmd/buildah/config.go

```diff
@@ -107,6 +107,63 @@ var (
 	}
 )
 
+func compareImageCmd(c *cli.Context, name string, cmd []string) (bool, error) {
+	if name == "" {
+		return false, nil
+	}
+	store, err := getStore(c)
+	if err != nil {
+		return false, err
+	}
+	ctx := getContext()
+	systemContext, err := parse.SystemContextFromOptions(c)
+	if err != nil {
+		return false, errors.Wrapf(err, "error building system context")
+	}
+	builder, err := openImage(ctx, systemContext, store, name)
+	if err != nil {
+		return false, errors.Wrapf(err, "error reading build object %q", name)
+	}
+	if len(builder.Cmd()) != len(cmd) {
+		return false, nil
+	}
+	for i, v := range builder.Cmd() {
+		if v != cmd[i] {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func updateEntrypoint(builder *buildah.Builder, c *cli.Context) {
+	if len(strings.TrimSpace(c.String("entrypoint"))) == 0 {
+		builder.SetEntrypoint(nil)
+	} else {
+		var entrypointJSON []string
+		err := json.Unmarshal([]byte(c.String("entrypoint")), &entrypointJSON)
+
+		if err != nil {
+			// it wasn't a valid json array, fall back to string
+			entrypointSpec := make([]string, 3)
+			entrypointSpec[0] = "/bin/sh"
+			entrypointSpec[1] = "-c"
+			entrypointSpec[2] = c.String("entrypoint")
+
+			builder.SetEntrypoint(entrypointSpec)
+		} else {
+			builder.SetEntrypoint(entrypointJSON)
+		}
+
+		equal, err := compareImageCmd(c, builder.FromImage, builder.Cmd())
+		if err != nil {
+			logrus.Errorf("error parsing cmd from %q: %v", builder.FromImage, err)
+		}
+		if equal {
+			builder.SetCmd(nil)
```

Comment:
Rather than setting the command to nil, could we just print out a warning with a recommendation to the user that they should set cmd to nil themselves?  That way we let them know of the potential issue, but don't disallow them from shooting themselves in the foot if they want to do so.

Reply:
Sure, do you have a description of the risk?  Can you set the cmd to nil, or only to ""?  Is there a difference?

Reply:
I think their foot might be alright...  

The Docker documentation has the following on Entrypoint:
> The shell form prevents any CMD or run command line arguments from being used

So if using shell form (a string) for ENTRYPOINT, it is going to ignore what's in CMD anyway, so they won't really shoot themselves in the foot.

If they use exec form (an array), existing CMD will be appended to the ENTRYPOINT on execution (which may cause them to shoot themselves in the foot if they have something like `/bin/bash` in CMD). But the way i've seen this used is ENTRYPOINT set in the parent, and then child just modifies CMD to adjust input parameters... but there may well be other use cases.

Example execution methods (punch line at the end!)

__exec form (uses CMD as parameter):__
```
FROM centos
ENTRYPOINT [ "/usr/bin/echo" ]
CMD [ "pixdrift" ]
```
Output
```
pixdrift
```

__shell form (ignores CMD value):__
```
FROM centos
ENTRYPOINT /usr/bin/echo
CMD [ "pixdrift" ]
```

_Produces no output_


So, if removal of CMD is taken out of `buildah config`.. error message could be

When using exec
> Warning: cmd 'pixdrift' exists and will be passed as parameter to entrypoint '/usr/bin/echo'

When using shell
> Warning: cmd 'pixdrift' exists but will be usurped by entrypoint

Probably don't need to leave 'usurped'.. late.. brain.. etc. 😄

Reply:
I'm good with @pixdrift's suggestions, perhaps with slight tweaks for late brain.  If we go this route, the title for this commit should change.